### ISSUE TITLE
Why no CancellationToken support?

### ISSUE THREAD

**Issue by ldematte** [influxdata/influxdb-client-csharp#173]
I am migrating from 1.7 to 2.0 and using this new library.
I noticed that WriteAsync calls do not accept a cancellation token. Is there a reason or is it simply missing/waiting for a PR?

**Comment by bednar** [comment]
Hi @ldematte, 

thanks for using our client. 

> Is there a reason or is it simply missing/waiting for a PR?

Unfortunately it is missing. Thanks for suggestion and We will take a look.

Regards

**Comment by ldematte** [comment]
Thanks for the quick answer. I have given it a look and it seems to be quite straightforward but long-ish. You have to drill down some levels to arrive to RestSharp ExecuteTaskAsync and then write overloads for all the public async methods to accept the CancellationToken. 
I am considering contributing with a PR but I am concerned about the second part in particular. Would it be OK if I do it only for WriteApi methods and let you/someone/myself else retrofit it to other APIs at a later stage?
Or would you prefer to do everything yourself?

**Comment by bednar** [comment]
> Would it be OK if I do it only for WriteApi methods and let you/someone/myself else retrofit it to other APIs at a later stage?

It is OK. We will fix rest of code later.

**Comment by bednar** [comment]
I've prepared #174 that introduces possibility to use `CancellationToken` in our REST API. You could use branch: `feat/cancellation-token-for-http` as a source for your PR.

**Comment by ldematte** [comment]
I have added my PR. I have tried to follow the guidelines, but let me know if something is wrong.
I have not added any test. I could add some tests that check that the correct CancellationToken is passed to the WriteService; with a minor change to the WriteApiAsync ctor from:
```cs
protected internal WriteApiAsync(InfluxDBClientOptions options, WriteService service, IDomainObjectMapper mapper, InfluxDBClient influxDbClient)
```
to
```cs
protected internal WriteApiAsync(InfluxDBClientOptions options, IWriteService service, IDomainObjectMapper mapper)
```
(remove unused InfluxDBClient, replace concrete WriteService  with interface) it would be possible to inejct a mock `IWriteService` to do the check.

I personally think those tests will add little value, but I would clean up the ctor anyway. What do you think?

**Comment by bednar** [comment]
> I personally think those tests will add little value, but I would clean up the ctor anyway. What do you think?

I agree, we don't need tests for this change

**Comment by bednar** [comment]
The issue is fixed in [1.17.0](https://github.com/influxdata/influxdb-client-csharp/milestone/20) milestone. You could use the preview version by:

```
dotnet add package InfluxDB.Client --version 1.17.0-dev.2619
```

https://www.nuget.org/packages/InfluxDB.Client/1.17.0-dev.2619
